packages/event-handler/src/rest/BaseRouter.ts

```diff
@@ -76,6 +105,68 @@ abstract class BaseRouter {
     }
   }
 
+  /**
+   * Handles errors by finding a registered error handler or falling
+   * back to a default handler.
+   *
+   * @param error - The error to handle
+   * @returns A Response object with appropriate status code and error details
+   */
+  protected handleError(error: Error): Response {
+    const handler = this.errorHandlerRegistry.resolve(error);
+    if (handler != null) {
+      try {
+        const body = handler(error);
+        return new Response(JSON.stringify(body), {
+          status: body.statusCode,
+          headers: { 'Content-Type': 'application/json' },
+        });
+      } catch (handlerError) {
+        return this.#defaultErrorHandler(handlerError as Error);
+      }
+    }
+
+    if (error instanceof ServiceError) {
+      return new Response(JSON.stringify(error.toJSON()), {
+        status: error.statusCode,
+        headers: { 'Content-Type': 'application/json' },
+      });
+    }
+
+    return this.#defaultErrorHandler(error);
+  }
+
+  /**
+   * Default error handler that returns a 500 Internal Server Error response.
+   * In development mode, includes stack trace and error details.
+   *
+   * @param error - The error to handle
+   * @returns A Response object with 500 status and error details
+   */
+  #defaultErrorHandler(error: Error): Response {
+    const isProduction =
```

Comment:
I don't think we should add this, we can just assume that if it's not dev mode, then it's prod. I don't know how many people actually set this

Reply:
Don't worry, impossible to make sure detailed issues without occasionally missing something. I much prefer not checking this variable because it makes it far harder for the handler to accidentally send unwanted details back in a response.